src/lib/vr.ts

```diff
@@ -152,3 +153,62 @@ export async function getStations() {
 
   return stations;
 }
+
+const trainsResponseSchema = z.array(
+  z.object({
+    trainNumber: z.number(),
+    trainType: z.string(),
+    timeTableRows: z.array(
+      z.object({
+        stationShortCode: z.string(),
+        scheduledTime: z.string(),
+      })
+    ),
+  })
+);
+
+export async function getInitialTrains(date: string) {
+  const [initialTrainsUnchecked, stations] = await Promise.all([
+    getJSON(`https://rata.digitraffic.fi/api/v1/trains/${date}`) as Promise<unknown>,
+    getStations(),
+  ]);
+
+  const initialTrains = trainsResponseSchema.parse(initialTrainsUnchecked);
+
+  return initialTrains
+    .filter((t) => ["IC", "S"].includes(t.trainType))
+    .map((t) => {
+      const departure = t.timeTableRows[0];
+      const arrival = t.timeTableRows[t.timeTableRows.length - 1];
+
+      if (!departure || !arrival) {
+        return {
+          value: t.trainNumber.toString(),
+          label: `${t.trainType}${t.trainNumber}`,
+          departureStationShortCode: "",
+          arrivalStationShortCode: "",
+          departureStationName: "",
+          arrivalStationName: "",
+        };
+      }
+
+      const departureTime = formatShortFinnishTime(departure.scheduledTime);
+      const arrivalTime = formatShortFinnishTime(arrival.scheduledTime);
+
+      const longDepartureStationName =
```

Comment:
What do you think about removing "asema" here? There were a few dozen stations that unnecessarily contained "asema", such as "Helsinki asema" or "Kerava asema"?

The only problematic case I found was "Pasila autojuna-asema", so it will get shortened to "Pasila autojuna-" which isn't very good.

Reply:
Looks like there's also "Lentoasema" (LEN) which is problematic

Reply:
We could simply replace only cases that contain a space, so ` asema`

Reply:
Yes, thats already what the code does in a couple of places, ex. https://github.com/Chicken/VenaaRauhassa/blob/master/src/pages/train/%5Bdate%5D/%5Btrain%5D.tsx#L413